### PR TITLE
Add `autoseek` feature when a payload binding fails or constraint validation fails

### PR DESCRIPTION
**Pull request by dilanSachi**
## Purpose
$subject
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/3827
Fixes https://github.com/ballerina-platform/ballerina-standard-library/issues/3884

## Checklist
- [x] Linked to an issue
- [x] Updated the changelog
- [x] Added tests
- [x] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: ballerina/kafka_records.bal [hunk 1]
```diff
@@ -163,6 +164,7 @@ public type ConsumerConfiguration record {|
     boolean excludeInternalTopics = true;
     boolean decoupleProcessing = false;
     boolean validation = true;
+    boolean autoSeek = true;
```
Comment (dilanSachi) [inline ballerina/kafka_records.bal:167]: autoSeekOnValidationError

File: ballerina/kafka_records.bal [hunk 2]
```diff
@@ -116,6 +116,7 @@ public type AuthenticationConfiguration record {|
 # + excludeInternalTopics - Whether records from internal topics should be exposed to the consumer
 # + decoupleProcessing - Decouples processing
 # + validation - Configuration related to constraint validation check
+# + autoSeekOnValidationFailure - Enables seeking past any errornous records found when data-binding and validating constraints
```
Comment (shafreenAnfar) [inline ballerina/kafka_records.bal:119]: Enables seeking past any errornous records when data-binding and validating constraints

Comment (shafreenAnfar) [reply ballerina/kafka_records.bal:119]: Automatically seeks past the  errornous records in the event of an data-binding or validating constraints failure

Comment (dilanSachi) [reply ballerina/kafka_records.bal]: Updated

**Review by shafreenAnfar — APPROVED** [review]

**Pull request merged**